File: src/main/java/com/thebluehats/server/game/world/StopLiquidFlow.java
package com.thebluehats.server.game.world;

import org.bukkit.Material;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockFromToEvent;

public class StopLiquidFlow implements Listener {
    @EventHandler
    public void onFlow(BlockFromToEvent event) {
        if (event.getBlock().getType() == Material.STATIONARY_WATER || event.getBlock().getType() == Material.STATIONARY_LAVA) {
            event.setCancelled(true);
        }
    }
}

File: minecraft-server/src/main/kotlin/com/thebluehats/server/game/world/StopLiquidFlow.kt
package com.thebluehats.server.game.world

import org.bukkit.Material
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.BlockFromToEvent

class StopLiquidFlow : Listener {
    @EventHandler
    fun onFlow(event: BlockFromToEvent) {
        if (event.block.type == Material.STATIONARY_WATER || event.block.type == Material.STATIONARY_LAVA) {
            event.isCancelled = true
        }
    }
}